it/src/test/java/com/google/cloud/teleport/it/spanner/DefaultSpannerResourceManagerTest.java

```diff
@@ -0,0 +1,395 @@
+            .set("LastName").to(string("Richards"))
+            .build();
+    Struct struct2 =
+        Struct.newBuilder()
+            .set("SingerId").to(int64(2))
+            .set("FirstName").to(string("Catalina"))
+            .set("LastName").to(string("Smith"))
+            .build();
+    // spotless:on
+    when(resultSet.getCurrentRowAsStruct()).thenReturn(struct1).thenReturn(struct2);
+    when(spanner.getDatabaseClient(any()).singleUse().read(any(), any(), any()))
+        .thenReturn(resultSet);
+
+    // act
+    ImmutableList<Struct> actual =
+        testManager.readTableRecords("Singers", "SingerId", "FirstName", "LastName");
+
+    // assert
+    ImmutableList<Struct> expected = ImmutableList.of(struct1, struct2);
+    assertThat(actual).containsExactlyElementsIn(expected);
+  }
+
+  @Test
+  public void testReadRecordsShouldThrowExceptionWhenSpannerReadFails()
+      throws ExecutionException, InterruptedException {
+    // arrange
+    prepareTable();
+    when(spanner.getDatabaseClient(any()).singleUse().read(any(), any(), any()))
+        .thenThrow(SpannerException.class);
+
+    // act & assert
+    assertThrows(
+        SpannerResourceManagerException.class,
+        () -> testManager.readTableRecords("Singers", "SingerId", "FirstName", "LastName"));
+  }
+
+  @Test
+  public void testCleanupAllShouldThrowExceptionWhenSpannerDeleteInstanceFails() {
+    // arrange
+    doThrow(SpannerException.class).when(instanceAdminClient).deleteInstance(any());
+    when(spanner.getInstanceAdminClient()).thenReturn(instanceAdminClient);
+    testManager = new DefaultSpannerResourceManager(spanner, TEST_ID, PROJECT_ID, REGION);
+
+    // act & assert
+    assertThrows(SpannerResourceManagerException.class, () -> testManager.cleanupAll());
+  }
+
+  @Test
+  public void testCleanupAllShouldWorkWhenSpannerDeleteInstanceSucceeds() {
+    // arrange
+    doNothing().when(instanceAdminClient).deleteInstance(any());
+    when(spanner.getInstanceAdminClient()).thenReturn(instanceAdminClient);
+    testManager = new DefaultSpannerResourceManager(spanner, TEST_ID, PROJECT_ID, REGION);
+
+    // act
+    testManager.cleanupAll();
+
+    // assert
+    verify(spanner.getInstanceAdminClient()).deleteInstance(any());
+    verify(spanner).close();
+  }
+
+  @Test
+  public void testManagerShouldBeUnusableAfterCleanup() {
+    // arrange
+    doNothing().when(instanceAdminClient).deleteInstance(any());
+    when(spanner.getInstanceAdminClient()).thenReturn(instanceAdminClient);
+    when(spanner.isClosed()).thenReturn(true);
+    testManager = new DefaultSpannerResourceManager(spanner, TEST_ID, PROJECT_ID, REGION);
+    testManager.cleanupAll();
+    String statement =
+        "CREATE TABLE Singers (\n"
+            + "  SingerId   INT64 NOT NULL,\n"
+            + "  FirstName  STRING(1024),\n"
+            + "  LastName   STRING(1024),\n"
+            + ") PRIMARY KEY (SingerId)";
+    Mutation testMutation =
```

Comment:
Please make this consistent with the other mutations. That'll require disabling spotless similarly to what was done with them.